unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/MyAccountHUD/MyProfileController.cs

```diff
@@ -0,0 +1,196 @@
+﻿using Cysharp.Threading.Tasks;
+using DCL.Browser;
+using DCL.Tasks;
+using DCLServices.Lambdas.NamesService;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using UnityEngine;
+
+namespace DCL.MyAccount
+{
+    public class MyProfileController
+    {
+        private const string CLAIM_UNIQUE_NAME_URL = "https://builder.decentraland.org/claim-name";
+
+        private readonly IMyProfileComponentView view;
+        private readonly DataStore dataStore;
+        private readonly IUserProfileBridge userProfileBridge;
+        private readonly INamesService namesService;
+        private readonly IBrowserBridge browserBridge;
+        private readonly MyAccountSectionHUDController myAccountSectionHUDController;
+        private readonly KernelConfig kernelConfig;
+        private readonly List<string> loadedNames = new ();
+
+        private CancellationTokenSource cts;
+        private Regex nameRegex;
+
+        private UserProfile ownUserProfile => userProfileBridge.GetOwn();
+        private string ownUserMainName => SplitUserName(ownUserProfile.userName).mainName;
+        private string ownUserNonClaimedHashtag => SplitUserName(ownUserProfile.userName).nonClaimedHashtag;
+
+        public MyProfileController(
+            IMyProfileComponentView view,
+            DataStore dataStore,
+            IUserProfileBridge userProfileBridge,
+            INamesService namesService,
+            IBrowserBridge browserBridge,
+            MyAccountSectionHUDController myAccountSectionHUDController,
+            KernelConfig kernelConfig)
+        {
+            this.view = view;
+            this.dataStore = dataStore;
+            this.userProfileBridge = userProfileBridge;
+            this.namesService = namesService;
+            this.browserBridge = browserBridge;
+            this.myAccountSectionHUDController = myAccountSectionHUDController;
+            this.kernelConfig = kernelConfig;
+
+            dataStore.myAccount.isMyAccountSectionVisible.OnChange += OnMyAccountSectionVisibleChanged;
+            view.OnCurrentNameEdited += OnNameEdited;
+            view.OnCurrentNameSubmitted += OnNameSubmitted;
+            view.OnGoFromClaimedToNonClaimNameClicked += GoFromClaimedToNonClaimName;
+            view.OnClaimNameClicked += OnClaimNameRequested;
+            ownUserProfile.OnUpdate += OnOwnUserProfileUpdated;
+
+            if (!Configuration.EnvironmentSettings.RUNNING_TESTS)
```

Comment:
we should avoid this.. you should be able to setup a `KernelConfig` in the tests (no need for extra interface)